### PR TITLE
Added img src/srcset to the source-set list update algorithm

### PR DESCRIPTION
**Pull request by yoavweiss**
As discussed on https://github.com/ResponsiveImagesCG/picture-element/issues/72 I've added img's src/srcset to the source-set list update algorithm. I'm not sure I did so in the most efficient manner, so a review would be appreciated.

I also get bikeshed errors, resulting from my use of img's srcset attribute.

### CODE REVIEW

File: index.src.html [hunk 1]
```diff
@@ -311,7 +310,7 @@ <h2 id='the-picture-element'>
 
 	Note: <a element>picture</a> is somewhat different from the similar-looking elements <a element>video</a> and <a element>audio</a>.
 	While all of them contain <a element>source</a> elements,
-	the syntax of the <a element-attr for="source">src</a> attribute is different when the element is nested within <a element>picture</a>,
+	the syntax of the <a element-attr for="source">srcset</a> attribute is different when the element is nested within <a element>picture</a>,
```
Comment (tabatkins) [inline index.src.html]: Good catch, but a simple rename isn't correct.  `<source>` in `<video>` doesn't use `srcset` at all - that's the difference we should be calling out here, that it pays attention to a different source-providing attribute when used inside of `<picture>`.

File: index.src.html [hunk 2]
```diff
@@ -396,8 +395,15 @@ <h4 id='update-source-sets'>
 
 			<ol>
 				<li>
-					If <var>child</var> is an <a element>img</a> element,
-					exit this sub-algorithm.
+					If <var>child</var> is an <a element>img</a> element and it has a <a element-attr for="img">srcset</a> attribute,
+                    <a title="parse a srcset attribute">parse <var>child</var>’s srcset attribute</a> and add the returned <a>source set</a>
```
Comment (tabatkins) [inline index.src.html]: Why the weird indentation change?

File: index.src.html [hunk 3]
```diff
@@ -496,6 +502,42 @@ <h4 id='parse-srcset-attr'>
 
 	Then return <var>source set</var>.
 
+<h4 id='parse-src-attr'>
+Parsing a <code>src</code> Attribute</h4>
+
+	When asked to <dfn title="parse a src attribute|parse its src attribute">parse a src attribute</dfn> from an element,
```
Comment (tabatkins) [inline index.src.html]: No need for grammar stuff here, plus that doesn't match how `src` is traditionally treated.  The value of `src` is just taken as a url, full stop.  Either HTML or the URL standard should describe how.

That means that it's just a matter of "if there's a `src` attribute, add [stuff] to the `list of source sets`".

File: index.src.html (hunk 2 above)
Comment (yoavweiss) [inline index.src.html]: Dunno. Will fix it.

File: index.src.html (hunk 1 above)
Comment (yoavweiss) [inline index.src.html]: What's the correct way to address this?

**Comment by tabatkins** [comment]
Looks good.

**Pull request merged**